apps/emqx/src/emqx_persistent_session_ds/emqx_persistent_session_ds_subs.erl

```diff
@@ -76,18 +78,45 @@
     superseded_by => subscription_state_id()
 }.
 
+-type new_stream_subs() :: #{emqx_ds_new_streams:watch() => emqx_types:topic()}.
+
 %%================================================================================
 %% API functions
 %%================================================================================
 
+-spec open(emqx_persistent_session_ds_state:t()) -> new_stream_subs().
+open(S) ->
+    %% Receive notifications about new streams for the topic filter:
+    fold_private_subscriptions(
+        fun(TopicFilter, _Sub, Acc) ->
+            Acc#{watch_streams(TopicFilter) => TopicFilter}
```

Comment:
Nit: it's a bit confusing where binary topic filters and already processed filters are used, specially since `emqx_ds:topic_filter() :: list(_)` and `emqx_persistent_session_ds:topic_filter() :: emqx_types:topic() | _SharedTopicFilter`.  From `watch_streams/1`, seems like the former is the case here.

```suggestion
            Acc#{watch_streams(TopicFilterBin) => TopicFilterBin}
```